src/java.desktop/macosx/classes/sun/java2d/macos/MacOSFlags.java

```diff
@@ -44,43 +52,122 @@
                 propString.equals("True") ||
                 propString.equals("T") ||
                 propString.equals("")) // having the prop name alone
             {                          // is equivalent to true
-                returnVal = true;
+                returnVal = PropertyState.ENABLED;
             } else if (propString.equals("false") ||
                        propString.equals("f") ||
                        propString.equals("False") ||
                        propString.equals("F"))
             {
-                returnVal = false;
+                returnVal = PropertyState.DISABLED;
             }
         }
         return returnVal;
     }
 
-
-    private static boolean getPropertySet(String p) {
+    private static boolean isBooleanPropTrueVerbose(String p) {
         String propString = System.getProperty(p);
-        return (propString != null) ? true : false;
+        if (propString != null) {
+            if (propString.equals("True") ||
+                propString.equals("T"))
+            {
+                return true;
+            }
+        }
+        return false;
     }
 
     private static void initJavaFlags() {
         java.security.AccessController.doPrivileged(
                 (PrivilegedAction<Object>) () -> {
-                    metalEnabled = getBooleanProp("sun.java2d.metal", false);
+                    PropertyState oglState = getBooleanProp("sun.java2d.opengl", PropertyState.UNSPECIFIED);
+                    PropertyState metalState = getBooleanProp("sun.java2d.metal", PropertyState.UNSPECIFIED);
+
+                    // Handle invalid combinations to use the default rendering pipeline
+                    // Current default rendering pipeline is OpenGL
+                    // (The default can be changed to Metal in future just by toggling two states in this if condition block)
+                    if ((oglState == PropertyState.UNSPECIFIED && metalState == PropertyState.UNSPECIFIED) ||
+                        (oglState == PropertyState.DISABLED && metalState == PropertyState.DISABLED) ||
+                        (oglState == PropertyState.ENABLED && metalState == PropertyState.ENABLED)) {
+                        oglState = PropertyState.ENABLED; // Enable default pipeline
+                        metalState = PropertyState.DISABLED; // Disable non-default pipeline
+                    }
+
+                    if (metalState == PropertyState.UNSPECIFIED) {
+                        if (oglState == PropertyState.DISABLED) {
+                            oglEnabled = false;
+                            metalEnabled = true;
+                        } else {
+                            oglEnabled = true;
+                            metalEnabled = false;
+                        }
+                    } else if (metalState == PropertyState.ENABLED) {
+                        if (oglState == PropertyState.ENABLED) {
+                            oglEnabled = true;
```

Comment:
By adding the invalid state handling further up, this condition can never be true

Reply:
Right. By the time we get here, the two flags cannot possibly be the same value, so you could simplify this by removing this check. That could be done as a follow-up.